### PR TITLE
tests: fix edge case where non-default python is used, by skipping it

### PR DESCRIPTION
**Pull request by eli-schwartz**
In a couple of python module tests, we try to test things that rely on the default python being the same one we look up in the python module. This is unsolvable for the deprecated python3 module, as it actually uses the in-process version of python for everything. For the python module, we could actually look up the default system python instead of the one we are running with, but then we wouldn't be testing the functionality of that alternative python... and also the install manifest tests would see files installed for the wrong version of python, and report a combination of missing+extra files...

Solve both tests by just skipping the parts we cannot check.

### CODE REVIEW

**Comment by eli-schwartz** [comment]
I discovered this locally running on a self-built python 3.11, and it turns out that one of the tests (not the cython one) is reproducing on the macOS CI jobs too.

/cc @xclaesse

**Comment by eli-schwartz** [comment]
Also a curious note. @carlocab, seems that /usr/local/bin/python3 is 3.11, but `/usr/local/bin/pkg-config --modversion python3` is 3.10?

(But I found this issue locally when python3 is 3.10, when running the testsuite as `python3.11 run_tests.py` -- it's an issue either way.)

**Comment by eli-schwartz** [comment]
@carlocab alright, bigger issue is genuinely an issue.

```
Installing collected packages: urllib3, idna, cython, coverage, charset-normalizer, requests, codecov
  WARNING: The scripts cygdb, cython and cythonize are installed in '/Library/Frameworks/Python.framework/Versions/3.11/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The scripts coverage, coverage-3.11 and coverage3 are installed in '/Library/Frameworks/Python.framework/Versions/3.11/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The script normalizer is installed in '/Library/Frameworks/Python.framework/Versions/3.11/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The script codecov is installed in '/Library/Frameworks/Python.framework/Versions/3.11/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
Successfully installed charset-normalizer-2.1.1 codecov-2.1.12 coverage-6.5.0 cython-0.29.32 idna-3.4 requests-2.28.1 urllib3-1.26.12
```

And attempting to run the codecov tool as the final stage of the CI run fails because command not found.

Also I guess the cython job only doesn't reproduce on macOS because it currently isn't found, so it gets skipped. :D

...

Why is the bin directory for pip install not in PATH? We could add it, but I don't really want to hardcode paths. Don't really want to pin the version either.

**Comment by eli-schwartz** [comment]
Alright, the python macOS issue is silly, as it turns out the base image scribbled all over homebrew despite literally installing homebrew python on its own. See https://github.com/Homebrew/discussions/discussions/3895

Should be fixed by having homebrew force relink itself.

**Comment by xclaesse** [comment]
LGTM if ci is green now.

**Comment by eli-schwartz** [comment]
Well, it worked a week ago. :D Now llvm is forcing python3 to 3.11, so the brew install fails before link --force can run. Hold on, reordering...

**Comment by jpakkane** [comment]
I'll merge this once the test goes green.

**Comment by eli-schwartz** [comment]
@carlocab is there some future-proof way to say "install whichever version of python is drawn in by llvm, and --force it, but don't --force any other packages"?

If not, I can always manually specify `brew install --force python@3.11` since I know externally that that is the version I need to fix CI, right?

**Comment by jpakkane** [comment]
```
 ==> Pouring python@3.11--3.11.0.big_sur.bottle.tar.gz
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /usr/local
```

:cry:

**Comment by eli-schwartz** [comment]
Yep. Just waiting on an idea of the cleanest way to do this before committing and pushing.

**Comment by carlocab** [comment]
> @carlocab is there some future-proof way to say "install whichever version of python is drawn in by llvm, and --force it, but don't --force any other packages"?

Something like
```
python=$(brew deps llvm | grep python)
brew install --force $python
brew install llvm
```
might do what you want here.

File: .github/workflows/macos.yml [hunk 1]
```diff
@@ -69,9 +69,15 @@ jobs:
     - uses: actions/checkout@v2
     # Avoid picking up an older version of LLVM that does not work.
     - run: brew update
+    # github actions overwrites brew's python. Force it to reassert itself, by running in a separate step.
+    - name: unbreak python in github actions
+      run: |
+        find /usr/local/bin -lname '*/Library/Frameworks/Python.framework/*' -delete
+        sudo rm -rf /Library/Frameworks/Python.framework/
+        brew install --force python3 && brew unlink python3 && brew unlink python3 && brew link --overwrite python3
```
Comment (carlocab) [inline .github/workflows/macos.yml:77]: It might suffice to just do
```
brew install --force --overwrite python3
```
here.
